src/overlays/actors/ovl_En_Goma/z_en_goma.c

```diff
@@ -73,77 +104,796 @@ static ColliderCylinderInit D_80A4B7CC = {
+    }
+}
+
+void EnGoma_EggFallToGround(EnGoma* this, GlobalContext* globalCtx) {
+    this->actor.gravity = -1.3f;
+    this->eggSquishAccel += 0.03f;
+    this->eggSquishAngle += 1.0f + this->eggSquishAccel;
+    Math_ApproachZeroF(&this->eggSquishAmount, 1.0f, 0.005f);
+    Math_ApproachF(&this->eggYOffset, 1500.0f, 1.0f, 150.0f);
+
+    switch (this->hatchState) {
+        case 0:
+            if (this->actor.bgCheckFlags & 1) { // floor
+                if (this->actor.params < 6) {
+                    Audio_PlayActorSound2(&this->actor, NA_SE_EN_GOMA_BJR_EGG1);
+                } else {
+                    Audio_PlayActorSound2(&this->actor, NA_SE_EN_GOMA_EGG1);
+                }
+
+                if (this->actor.params > 5) {
+                    EnGoma_SetupHatch(this, globalCtx);
+                } else {
+                    this->hatchState = 1;
+                    this->actionTimer = 3;
+                    Math_ApproachF(&this->eggScale, 1.5f, 0.5f, 1.0f);
+                }
+            }
+            break;
+
+        case 1:
+            if (this->actionTimer == 0) {
+                this->hatchState = 2;
+                this->actionTimer = 3;
+                Math_ApproachF(&this->eggScale, 0.75f, 0.5f, 1.0f);
+                this->actor.velocity.y = 5.0f;
+                this->actor.speedXZ = 2.0f;
+            } else {
+                Math_ApproachF(&this->eggScale, 1.5f, 0.5f, 1.0f);
+            }
+            break;
+
+        case 2:
+            if (this->actionTimer == 0) {
+                this->hatchState = 3;
+                this->actionTimer = 80;
+            } else {
+                Math_ApproachF(&this->eggScale, 0.75f, 0.5f, 1.0f);
+            }
+            break;
+
+        case 3:
+            Math_ApproachF(&this->eggScale, 1.0f, 0.1f, 0.1f);
+            if (this->actionTimer == 0) {
+                EnGoma_SetupHatch(this, globalCtx);
+            }
+            break;
+    }
+
+    if (this->actor.bgCheckFlags & 1) {
+        Math_ApproachZeroF(&this->actor.speedXZ, 0.2f, 0.05f);
+    }
+    this->eggPitch += (this->actor.speedXZ * 0.1f);
+    this->actor.shape.rot.y = this->actor.world.rot.y;
+}
+
+void EnGoma_Egg(EnGoma* this, GlobalContext* globalCtx) {
+    Player* player = PLAYER;
+    s32 i;
+
+    this->eggSquishAngle += 1.0f;
+    Math_ApproachF(&this->eggSquishAmount, 0.1f, 1.0f, 0.005f);
+    if (fabsf(this->actor.world.pos.x - player->actor.world.pos.x) < 100.0f &&
+        fabsf(this->actor.world.pos.z - player->actor.world.pos.z) < 100.0f) {
+        if (++this->playerDetectionTimer > 9) {
+            this->actionFunc = EnGoma_EggFallToGround;
+        }
+    } else {
+        this->playerDetectionTimer = 0;
+    }
+
+    if (!(this->eggTimer & 0xF) && Rand_ZeroOne() < 0.5f) {
+        for (i = 0; i < 2; i++) {
+            Vec3f vel = { 0.0f, 0.0f, 0.0f };
+            Vec3f acc = { 0.0f, -0.5f, 0.0f };
+            Vec3f pos;
+
+            // lol poop
```

Comment:
Im assuming this wasn't meant to be left in lol.

Reply:
looool yeah forgot about that one